datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -330,6 +331,12 @@ impl PhysicalExpr for InListExpr {
             self.static_filter.clone(),
         )))
     }
+
+    fn dyn_hash(&self, _state: &mut dyn Hasher) {
```

Comment:
The same comment from above here applies -- I think a less precise hash function (with more collisions) is better than a runtime panic. Thus I think we should implement `Hash` for `static_filter`